packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -0,0 +1,277 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../android/android_sdk.dart';
+import '../base/common.dart';
+import '../base/context.dart';
+import '../base/file_system.dart';
+import '../base/os.dart';
+import '../base/platform.dart';
+import '../base/terminal.dart';
+import '../base/utils.dart';
+import '../base/version.dart';
+import '../build_info.dart';
+import '../cache.dart';
+import '../globals.dart';
+import '../project.dart';
+import '../reporting/reporting.dart';
+import 'android_sdk.dart';
+import 'android_studio.dart';
+
+/// The environment variables needed to run Gradle.
+Map<String, String> get gradleEnv {
+  final Map<String, String> env = Map<String, String>.from(platform.environment);
+  if (javaPath != null) {
+    // Use java bundled with Android Studio.
+    env['JAVA_HOME'] = javaPath;
+  }
+  // Don't log analytics for downstream Flutter commands.
+  // e.g. `flutter build bundle`.
+  env['FLUTTER_SUPPRESS_ANALYTICS'] = 'true';
+  return env;
+}
+
+/// Gradle utils in the current [AppContext].
+GradleUtils get gradleUtils => context.get<GradleUtils>();
+
+/// Provides utilities to run a Gradle task,
+/// such as finding the Gradle executable or constructing a Gradle project.
+class GradleUtils {
+  /// Empty constructor.
+  GradleUtils();
+
+  /// Gets the Gradle executable path and prepares the Gradle project.
+  /// This is the `gradlew` or `gradlew.bat` script in the `android/` directory.
+  String getExecutable(FlutterProject project) {
+    final Directory androidDir = project.android.hostAppGradleRoot;
+    // Update the project if needed.
+    // TODO(egarciad): https://github.com/flutter/flutter/issues/40460.
+    migrateToR8(androidDir);
+    injectGradleWrapperIfNeeded(androidDir);
+
+    final File gradle = androidDir.childFile(
+      platform.isWindows ? 'gradlew.bat' : 'gradlew',
+    );
+    if (gradle.existsSync()) {
+      printTrace('Using gradle from ${gradle.absolute.path}.');
+      return gradle.absolute.path;
+    }
+    throwToolExit('Unable to locate gradlew script');
```

Comment:
this error message should be more actionable (what should people do in response?)